app/pages/main/main-page.component.js

```diff
@@ -1,4 +1,34 @@
+class CategoryController {
+    constructor(mainPageService) {
+        'ngInject';
+        this.mainPageService = mainPageService;
+    }
+
+    $onInit() {
+        this.getProducts();
+        this.data = [];
+        this.newData = [];
+    }
+
+    getProducts() {
+        this.mainPageService.getProducts().then((data) => {
+            this.data = data;
+
+            this.sortArray(this.data, this.newData);
```

Comment:
It would be better to move logic into the service, so maybe create a function called `getPromotedProducts()` or something similar and call it